src/vs/workbench/contrib/notebook/browser/contrib/statusBar/contributedStatusBarItemController.ts

```diff
@@ -35,9 +35,9 @@ export class ContributedStatusBarItemController extends Disposable implements IN
 	}
 
 	private _updateEverything(): void {
-		this._visibleCells.forEach(cell => cell.dispose());
-		this._visibleCells.clear();
-		this._updateVisibleCells({ added: this._observer.visibleCells, removed: [] });
+		const newCells = this._observer.visibleCells.filter(cell => !this._visibleCells.has(cell.handle));
+		this._visibleCells.forEach(helper => helper.update());
```

Comment:
Update existing helpers
& create new helpers for any of the new cells (i.e. add helpers for cells that we haven't already tracked).

Reply:
It might be better to not update the ones that are about to be removed? We will call the provide method then immediately cancel it right?